indico/modules/users/operations.py

```diff
@@ -94,3 +94,37 @@ def delete_or_anonymize_user(user):
     else:
         signals.users.db_deleted.send(user, flushed=True)
         logger.info('User %r deleted %s', session.user if session else None, user_repr)
+
+
+def grant_admin(user, *, extra_log_data=None):
+    if user.is_admin:
+        return
+    user.is_admin = True
+    session_user = session.user if session else None
+    log_data = dict(extra_log_data or {})
+    if remote_addr := (request.remote_addr if request else None):
+        log_data['IP'] = remote_addr
+    AppLogEntry.log(AppLogRealm.admin, LogKind.positive, 'Admins',
+                    f'Admin privileges granted to {user.full_name}', session_user,
+                    data={**log_data, 'User ID': user.id})
```

Comment:
Just thinking... would it make sense to also add `meta={'user_id': user.id}`? This would allow filtering global log entries by ones related to that user (e.g. you'd see both the time they were granted and revoked admin rights)

Reply:
OTOH you can just go to the user log when you want more details...